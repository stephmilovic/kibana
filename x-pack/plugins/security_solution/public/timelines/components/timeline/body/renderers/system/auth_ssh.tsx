/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';

import { DraggableBadge } from '../../../../../../common/components/draggables';
import { TokensFlexItem } from '../helpers';

interface Props {
  contextId: string;
  eventId: string;
  sshSignature: string | null | undefined;
  sshMethod: string | null | undefined;
  timelineId: string;
}

export const AuthSsh = React.memo<Props>(
  ({ contextId, eventId, sshSignature, sshMethod, timelineId }) => (
    <>
      {sshSignature != null && (
        <TokensFlexItem grow={false} component="span">
          <DraggableBadge
            contextId={contextId}
            eventId={eventId}
            field="system.audit.package.name"
            value={sshSignature}
            iconType="document"
            timelineId={timelineId}
          />
        </TokensFlexItem>
      )}
      {sshMethod != null && (
        <TokensFlexItem grow={false} component="span">
          <DraggableBadge
            contextId={contextId}
            eventId={eventId}
            field="system.audit.package.version"
            value={sshMethod}
            iconType="document"
            timelineId={timelineId}
          />
        </TokensFlexItem>
      )}
    </>
  )
);

AuthSsh.displayName = 'AuthSsh';
