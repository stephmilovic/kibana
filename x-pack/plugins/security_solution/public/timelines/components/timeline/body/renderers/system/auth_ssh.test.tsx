/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { shallow } from 'enzyme';
import React from 'react';

import { AuthSsh } from './auth_ssh';

describe('AuthSsh', () => {
  describe('rendering', () => {
    test('it renders against shallow snapshot', () => {
      const wrapper = shallow(
        <AuthSsh
          contextId="[context-123]"
          eventId="[event-123]"
          sshSignature="[ssh-signature]"
          sshMethod="[ssh-method]"
          timelineId="timeline-1"
        />
      );
      expect(wrapper).toMatchSnapshot();
    });

    test('it returns null if sshSignature and sshMethod are both null', () => {
      const wrapper = shallow(
        <AuthSsh
          contextId="[context-123]"
          eventId="[event-123]"
          sshSignature={null}
          sshMethod={null}
          timelineId="timeline-1"
        />
      );
      expect(wrapper.children().length).toEqual(0);
    });

    test('it returns null if sshSignature and sshMethod are both undefined', () => {
      const wrapper = shallow(
        <AuthSsh
          contextId="[context-123]"
          eventId="[event-123]"
          sshSignature={undefined}
          sshMethod={undefined}
          timelineId="timeline-1"
        />
      );
      expect(wrapper.children().length).toEqual(0);
    });

    test('it returns null if sshSignature is null and sshMethod is undefined', () => {
      const wrapper = shallow(
        <AuthSsh
          contextId="[context-123]"
          eventId="[event-123]"
          sshSignature={null}
          sshMethod={undefined}
          timelineId="timeline-1"
        />
      );
      expect(wrapper.children().length).toEqual(0);
    });

    test('it returns null if sshSignature is undefined and sshMethod is null', () => {
      const wrapper = shallow(
        <AuthSsh
          contextId="[context-123]"
          eventId="[event-123]"
          sshSignature={undefined}
          sshMethod={null}
          timelineId="timeline-1"
        />
      );
      expect(wrapper.children().length).toEqual(0);
    });

    test('it returns sshSignature if sshMethod is null', () => {
      const wrapper = shallow(
        <AuthSsh
          contextId="[context-123]"
          eventId="[event-123]"
          sshSignature="[sshSignature-1]"
          sshMethod={null}
          timelineId="timeline-1"
        />
      );
      expect(wrapper.find('DraggableBadge').prop('value')).toEqual('[sshSignature-1]');
    });

    test('it returns sshMethod if sshSignature is null', () => {
      const wrapper = shallow(
        <AuthSsh
          contextId="[context-123]"
          eventId="[event-123]"
          sshSignature={null}
          sshMethod="[sshMethod-1]"
          timelineId="timeline-1"
        />
      );
      expect(wrapper.find('DraggableBadge').prop('value')).toEqual('[sshMethod-1]');
    });
  });
});
