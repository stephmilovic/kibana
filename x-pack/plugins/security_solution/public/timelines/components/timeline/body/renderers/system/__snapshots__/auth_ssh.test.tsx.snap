// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthSsh rendering it renders against shallow snapshot 1`] = `
<Fragment>
  <TokensFlexItem
    component="span"
    grow={false}
  >
    <DraggableBadge
      contextId="[context-123]"
      eventId="[event-123]"
      field="system.audit.package.name"
      iconType="document"
      timelineId="timeline-1"
      value="[ssh-signature]"
    />
  </TokensFlexItem>
  <TokensFlexItem
    component="span"
    grow={false}
  >
    <DraggableBadge
      contextId="[context-123]"
      eventId="[event-123]"
      field="system.audit.package.version"
      iconType="document"
      timelineId="timeline-1"
      value="[ssh-method]"
    />
  </TokensFlexItem>
</Fragment>
`;
